external/js/cothority/src/personhood/spawner-instance.ts

```diff
@@ -124,34 +124,45 @@ export default class SpawnerInstance extends Instance {
     }
 
     /**
-     * Spawns one or more darc and returns an array of all the spawned darcs.
+     * Create the instructions necessary to spawn one or mroe darcs. This is separated from the
+     * spanDarcs method itself, so that a caller can create a bigger ClientTransaction with
+     * multiple sets of instructions inside.
      *
-     * @param coin      The coin instance to take coins from
-     * @param signers   The signers for the transaction
-     * @param darcs... All the darcs to spawn using the spawner. The coin instance must have enough
-     * coins to pay for all darcs.
-     * @returns a promise that resolves with the new array of the instantiated darc instances
+     * @param coin where to take the coins from
+     * @param darcs the darcs to create
      */
-    async spawnDarcs(coin: CoinInstance, signers: Signer[], ...darcs: Darc[]): Promise<DarcInstance[]> {
+    spawnDarcInstructions(coin: CoinInstance, ...darcs: Darc[]): Instruction[] {
         const cost = this.struct.costDarc.value.mul(darcs.length);
-        const ctx = new ClientTransaction({
-            instructions: [
-                Instruction.createInvoke(
-                    coin.id,
-                    CoinInstance.contractID,
-                    CoinInstance.commandFetch,
-                    [new Argument({name: CoinInstance.argumentCoins, value: Buffer.from(cost.toBytesLE())})],
-                ),
-            ],
-        });
+        const ret: Instruction[] = [
+            Instruction.createInvoke(
+                coin.id,
+                CoinInstance.contractID,
+                CoinInstance.commandFetch,
+                [new Argument({name: CoinInstance.argumentCoins, value: Buffer.from(cost.toBytesLE())})],
+            ),
```

Comment:
```javascript
...darcs.map(...)
```

Reply:
Starts to feel too functional for me... I miss go ;)